File: aidp-java-client/src/main/java/com/oracle/aidataplatform/dp/model/GitRepairWorkTreeDetails.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;



/**
 * Details needed to repair git work tree for a folder context.
**/
@jakarta.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=GitRepairWorkTreeDetails.Builder.class)

public final class GitRepairWorkTreeDetails  {
    @Deprecated
    @java.beans.ConstructorProperties({"gitFolderPath", "branchName"})
    public GitRepairWorkTreeDetails(String gitFolderPath, String branchName) {
        super();
        this.gitFolderPath = gitFolderPath;
        this.branchName = branchName;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
                /**
     * Folder path used to locate the git worktree.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("gitFolderPath")
private String gitFolderPath;

        /**
         * Folder path used to locate the git worktree.
         * @param gitFolderPath the value to set
         * @return this builder
         **/
        

public Builder gitFolderPath(String gitFolderPath) {
    this.gitFolderPath = gitFolderPath;
    return this;
}
            /**
     * The name of the git branch
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("branchName")
private String branchName;

        /**
         * The name of the git branch
         * @param branchName the value to set
         * @return this builder
         **/
        

public Builder branchName(String branchName) {
    this.branchName = branchName;
    return this;
}


        public GitRepairWorkTreeDetails build() {
            GitRepairWorkTreeDetails model = new GitRepairWorkTreeDetails(this.gitFolderPath
                , this.branchName);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(GitRepairWorkTreeDetails model) {
                this.gitFolderPath(model.getGitFolderPath());
    this.branchName(model.getBranchName());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * Folder path used to locate the git worktree.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("gitFolderPath")
    private final String gitFolderPath;

        /**
     * Folder path used to locate the git worktree.
     * @return the value
     **/
    
    public String getGitFolderPath() {
        return gitFolderPath;
    }


        /**
     * The name of the git branch
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("branchName")
    private final String branchName;

        /**
     * The name of the git branch
     * @return the value
     **/
    
    public String getBranchName() {
        return branchName;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("GitRepairWorkTreeDetails(");
        sb.append("gitFolderPath=").append(String.valueOf(this.gitFolderPath));
        sb.append(", branchName=").append(String.valueOf(this.branchName));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof GitRepairWorkTreeDetails)) {
            return false;
        }

        GitRepairWorkTreeDetails other = (GitRepairWorkTreeDetails) o;
        return java.util.Objects.equals(this.gitFolderPath, other.gitFolderPath) &&
            java.util.Objects.equals(this.branchName, other.branchName);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.gitFolderPath == null ? 43 : this.gitFolderPath.hashCode());
        result = (result * PRIME) + (this.branchName == null ? 43 : this.branchName.hashCode());
        return result;
    }


}

File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/git_repair_work_tree_details.py
# coding: utf-8
# Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.



from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class GitRepairWorkTreeDetails(object):
    """
    Details needed to repair Git work tree for a folder context.
    """

    def __init__(self, **kwargs):
        """
        Initializes a new GitRepairWorkTreeDetails object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param git_folder_path:
            The value to assign to the git_folder_path property of this GitRepairWorkTreeDetails.
        :type git_folder_path: str

        :param branch_name:
            The value to assign to the branch_name property of this GitRepairWorkTreeDetails.
        :type branch_name: str

        """
        self.swagger_types = {
            'git_folder_path': 'str',
            'branch_name': 'str'
        }

        self.attribute_map = {
            'git_folder_path': 'gitFolderPath',
            'branch_name': 'branchName'
        }

        self._git_folder_path = None
        self._branch_name = None

    @property
    def git_folder_path(self):
        """
        **[Required]** Gets the git_folder_path of this GitRepairWorkTreeDetails.
        Folder path used to locate the Git worktree.


        :return: The git_folder_path of this GitRepairWorkTreeDetails.
        :rtype: str
        """
        return self._git_folder_path

    @git_folder_path.setter
    def git_folder_path(self, git_folder_path):
        """
        Sets the git_folder_path of this GitRepairWorkTreeDetails.
        Folder path used to locate the Git worktree.


        :param git_folder_path: The git_folder_path of this GitRepairWorkTreeDetails.
        :type: str
        """
        self._git_folder_path = git_folder_path

    @property
    def branch_name(self):
        """
        **[Required]** Gets the branch_name of this GitRepairWorkTreeDetails.
        The name of the Git branch.


        :return: The branch_name of this GitRepairWorkTreeDetails.
        :rtype: str
        """
        return self._branch_name

    @branch_name.setter
    def branch_name(self, branch_name):
        """
        Sets the branch_name of this GitRepairWorkTreeDetails.
        The name of the Git branch.


        :param branch_name: The branch_name of this GitRepairWorkTreeDetails.
        :type: str
        """
        self._branch_name = branch_name

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other
